app/assets/javascripts/ordered_creator.js

```diff
@@ -1,12 +1,45 @@
+function bindUpDownArrow() {
+  $('.ordered-up-arrow').unbind("click").on('click', function(e) {
+    swapUp(this);
+  });
+  $('.ordered-down-arrow').unbind("click").on('click', function(e) {
+    swapDown(this);
+  });
+}
+
+function swapUp(selector) {
+  $(selector).parent().insertBefore($(selector).parent().prev());
+  reindexNestedOrderedField();
+}
+
+function swapDown(selector) {
+  $(selector).parent().insertAfter($(selector).parent().next());
+  reindexNestedOrderedField();
+}
+
 function orderedDragAndDrop(selector) {
-  selector.nestable({maxDepth: 0, rootClass: 'ordered-field-container'});
-  selector.on('change', function(event) {
-    hidden_index_selectors = $(this).find(".index");
-    hidden_index_selectors.each(function( index ) {
-      hidden_index_selectors[index].value = index;
-    });
+  selector.nestable({ maxDepth: 0, rootClass: 'ordered-field-container' });
+  selector.on('change', function (event) {
+    reindexNestedOrderedField();
+  });
+}
+
+function reindexNestedOrderedField(mutationsList) {
+  hidden_index_selectors = $($(".ordered-field-container .index"));
```

Comment:
syntax issue is still here

Reply:
This should be fixed with the merge of my other PR which was updated. Ill rebase this one on that after.